### PR TITLE
Bump AGP to 7.2.1 and Gradle to 7.5.0

### PR DESCRIPTION
**Pull request by adinauer**
## :bulb: Motivation and Context

Use latest AGP version as default.
Test against latest versions.

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt [hunk 1]
```diff
@@ -99,10 +99,14 @@ abstract class BaseSentryPluginTest(
             arrayOf("7.0.4", "7.2"),
             arrayOf("7.1.2", "7.2"),
             arrayOf("7.1.2", "7.3.3"),
-            arrayOf("7.1.3", "7.4"),
-            arrayOf("7.2.0", "7.4"),
-            arrayOf("7.3.0-beta01", "7.4"),
-            arrayOf("7.4.0-alpha01", "7.4.2")
+            arrayOf("7.1.3", "7.2"),
+            arrayOf("7.1.3", "7.5"),
```
Comment (marandaneto) [inline plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt:103]: We don't need to test with every patch from AGP.
Just change `7.1.2` to `7.1.3` and add a new case for Gradle 7.4 and 7.5

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt [hunk 2]
```diff
@@ -99,10 +99,14 @@ abstract class BaseSentryPluginTest(
...
+            arrayOf("7.2.1", "7.3.3"),
+            arrayOf("7.2.1", "7.5"),
```
Comment (marandaneto) [inline plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt:105]: Missing Gradle 7.4?

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt [hunk 3]
```diff
@@ -99,10 +99,14 @@ abstract class BaseSentryPluginTest(
...
+            arrayOf("7.3.0-beta05", "7.3.3"),
+            arrayOf("7.3.0-beta05", "7.5"),
+            arrayOf("7.4.0-alpha08", "7.3.3."),
+            arrayOf("7.4.0-alpha08", "7.5")
```
Comment (marandaneto) [inline plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt:109]: Alphas we can test only the latest Gradle version, so here in both cases would be `7.3.0-beta05` and 7.5, `7.4.0-alpha08` and 7.5

File: .github/workflows/test-publish-dry-run.yaml [hunk 4]
```diff
@@ -29,14 +29,29 @@ jobs:
           - agp: "7.1.2"
             gradle: "7.4"
             java: "11"
-          - agp: "7.2.0"
-            gradle: "7.4"
+          - agp: "7.1.3"
```
Comment (marandaneto) [inline .github/workflows/test-publish-dry-run.yaml:32]: Same comments from https://github.com/getsentry/sentry-android-gradle-plugin/pull/363/files#r936589706

Comment (adinauer) [reply .github/workflows/test-publish-dry-run.yaml:29]: updated

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt (hunk 3 above)
Comment (adinauer) [reply plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt]: updated

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt (hunk 2 above)
Comment (adinauer) [reply plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt]: updated

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt (hunk 1 above)
Comment (adinauer) [reply plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt]: updated

**Comment by adinauer** [comment]
Build is broken as of AGP 7.4.0-alpha04, works with 7.4.0-alpha03. Created https://github.com/getsentry/sentry-android-gradle-plugin/issues/364 to track.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**